pkg/agent/nodeportlocal/testing/annotations.go

```diff
@@ -40,24 +40,17 @@ func NewExpectedNPLAnnotations(nodeIP *string, nplStartPort, nplEndPort int) *Ex
 	}
 }
 
-func (a *ExpectedNPLAnnotations) find(podPort int) *nplk8s.NPLAnnotation {
+func (a *ExpectedNPLAnnotations) find(podPort int, protocol string) *nplk8s.NPLAnnotation {
 	for _, annotation := range a.annotations {
-		if annotation.PodPort == podPort {
+		if annotation.PodPort == podPort && annotation.Protocol == protocol {
 			return &annotation
 		}
 	}
 	return nil
 }
 
-func (a *ExpectedNPLAnnotations) Add(nodePort *int, podPort int, protocols ...string) *ExpectedNPLAnnotations {
-	for i, annotation := range a.annotations {
-		if annotation.PodPort == podPort {
-			annotation.Protocols = append(annotation.Protocols, protocols...)
-			a.annotations[i] = annotation
-			return a
-		}
-	}
-	annotation := nplk8s.NPLAnnotation{PodPort: podPort, Protocols: protocols}
+func (a *ExpectedNPLAnnotations) Add(nodePort *int, podPort int, protocol string, protocols ...string) *ExpectedNPLAnnotations {
```

Comment:
Not sure why we are keeping the `protocols` parameter here. It should just be `protocol` and you can generate the `protocols` field based on that if needed.

Reply:
Yes, protocol string is enough as input parameter, updated.